### DISCUSSION TITLE
WSL detection issue during installation

### DISCUSSION THREAD

**Discussion by kele14x** [AMDResearch/Riallto#14] [General]
Hi,

I notice that the installer will detect if WSL is installed on machine by checking the **Windows Subsystem for Linux** feature in **Windows Optional Features**. However the latest WSL is distributed via Microsoft Store and does not require **Windows Subsystem for Linux** feature enabled. So even I have WSL installed, the installer still ask me to enable it in Windows Optional Features.

Is there a more proper way to detect WSL installation?

**Comment by skalade** [comment]
Hi @kele14x,

Are you still able to run WSL Ubuntu distros without that feature enabled? I found that in practice even if I had WSL installed I still had trouble bringing up a Ubuntu instance without them (infamous error 0x80370114). We have documentation on enabling required features here: https://riallto.ai/prerequisites-wsl.html#prerequisites-wsl and recommend enabling them before install.

Thanks
Shawn

**Reply by kele14x** [reply]
Hi Shawn,

Yes seems from a specific version this could be possible:

![image](https://github.com/AMDResearch/Riallto/assets/4020067/36cbb5c8-57a4-4fdd-9c8a-4e106467d9f5)

As you can seen form the screen shoot, the WSL (WSL2) is running without **Windows Subsystem for Linux** enabled. Not sure about the Windows or WSL version requirement.

I'm running Windows 11 Version 23H2 (OS Build 22631.3155) and `wsl --version` output:

```
WSL version: 2.0.14.0
Kernel version: 5.15.133.1-1
WSLg version: 1.0.59
MSRDC version: 1.2.4677
Direct3D version: 1.611.1-81528511
DXCore version: 10.0.25131.1002-220531-1700.rs-onecore-base2-hyp
Windows version: 10.0.22631.3155
```
